speakeasy/winenv/api/usermode/kernel32.py

```diff
@@ -5567,3 +5567,158 @@ def AddVectoredExceptionHandler(self, emu, argv, ctx={}):
+        if res is None:
+            return 0
+
+        hnd = self.get_handle()
+        self.find_resources.update({hnd: {"ptr": pe.get_base() + res.data.struct.OffsetToData,
+                                          "size": res.data.struct.Size}})
+        return hnd
+
+    @apihook("GetUserDefaultLCID", argc=0)
+    def GetUserDefaultLCID(self, emu, argv, ctx={}):
+        '''
+        LCID GetUserDefaultLCID();
+        '''
+        # https://docs.microsoft.com/en-us/windows/win32/intl/locale-user-default
+        return int("0x0400", 16)
+
+    @apihook("GetTempFileNameW", argc=4)
+    def GetTempFileNameW(self, emu, argv, ctx={}):
+        '''
+        UINT GetTempFileNameW(
+            [in]  LPCWSTR lpPathName,
+            [in]  LPCWSTR lpPrefixString,
+            [in]  UINT    uUnique,
+            [out] LPWSTR  lpTempFileName
+        );
+        '''
+        lpPathName, lpPrefixString, uUnique, lpTempFileName = argv
+
+        cw = self.get_char_width(ctx)
+        path = self.read_mem_string(lpPathName, cw)
+        prefix = self.read_mem_string(lpPrefixString, cw)
+
+        import time
+        if prefix:
+            out = path + f"\\{prefix}_{int(time.time())}.tmp"
+        else:
+            out = path + f"\\{prefix}_{int(time.time())}.tmp"
+        argv[1] = out
+        self.write_mem_string(out, lpTempFileName, cw)
+
+        return len(out) + 1
+
+    @apihook('_llseek', argc=3)
+    def _llseek(self, emu, argv, ctx={}):
+        """
+        LONG _llseek(
+            HFILE hFile,
+            LONG  lOffset,
+            int   iOrigin
+        );
+        """
+        hFile, lOffset, iOrigin = argv
+        rv = 0
+
+        f = self.file_get(hFile)
+        if f:
+            rv = f.tell()
+            emu.set_last_error(windefs.ERROR_SUCCESS)
+
+        return rv
+
+    @apihook("_lopen", argc=2)
+    def _lopen(self, emu, argv, ctx={}):
+        '''
+        HFILE _lopen(
+            LPCSTR lpPathName,
+            int    iReadWrite
+        );
+        '''
+        lpFileName, iRedWrite = argv
+        cw = self.get_char_width(ctx)
+        filename = self.read_mem_string(lpFileName, cw)
+        fHandle = self.file_open(filename)
+        return fHandle
+
+    @apihook('_lclose', argc=1)
+    def _lclose(self, emu, argv, ctx={}):
+        '''
+        HFILE _lclose(
+            HFILE hFile
+            );
+        '''
+        hObject, = argv
+        obj = self.get_object_from_handle(hObject)
+        if obj:
+            emu.dec_ref(obj)
+            return True
+        return False
+
+    @apihook('GetConsoleTitle', argc=2)
+    def GetConsoleTitle(self, emu, argv, ctx={}):
+        '''
+        DWORD WINAPI GetConsoleTitle(
+            _Out_ LPTSTR lpConsoleTitle,
+            _In_  DWORD  nSize
+        ); 
+        '''   
+        #lpBuffer, nSize = argv
+        lpConsoleTitle, nSize = argv
+        cw = self.get_char_width(ctx)
+        rv = False
+
+        example_title = "explorer.exe"
```

Comment:
i think this is reasonable, though perhaps eventually we'd want to make this configurable, such as how the image path is configured here:

https://github.com/mandiant/speakeasy/blob/319121e9627a27429067546cd378cbfeed07f431/speakeasy/winenv/api/usermode/kernel32.py#L480-L483